src/AzureAuth/CommandInfo.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.AzureAuth
+{
+    using System.IO.Abstractions;
+    using System.Reflection;
+    using McMaster.Extensions.CommandLineUtils;
+    using Microsoft.Extensions.Logging;
+    using Microsoft.Office.Lasso.Telemetry;
+
+    /// <summary>
+    /// The command <see cref="CommandInfo"/> shows debug information and system runtime.
+    /// </summary>
+    [Command(Name = "info", Description = "Show debug information of AzureAuth. Please provide when asking for help.")]
+    internal class CommandInfo
+    {
+        private const string OptionResetDeviceID = "--reset-device-id";
+
+        /// <summary>
+        /// Gets or sets a value indicating whether reset device id.
+        /// </summary>
+        [Option(OptionResetDeviceID, "Reset Device ID", CommandOptionType.NoValue)]
+        public bool ResetDeviceID { get; set; }
+
+        /// <summary>
+        /// This method executes the info process.
+        /// </summary>
+        /// <param name="logger">The logger.</param>
+        /// <param name="fileSystem">The file system.</param>
+        /// <returns>The error code: 0 is normal execution, and the rest means errors during execution.</returns>
+        public int OnExecute(ILogger<CommandInfo> logger, IFileSystem fileSystem)
+        {
+            if (this.ResetDeviceID)
+            {
+                return this.ResetID(logger, fileSystem);
+            }
+
+            Assembly assembly = Assembly.GetExecutingAssembly();

```

Comment:
Can this ever return a null?

Reply:
Nope. Different with `GetEntryAssembly`, `GetExecutingAssembly` will return the assembly where it is invoked. Because AzureAuth is a .Net application, `GetExecutingAssembly` will always return a value.